crates/bevy_winit/src/winit_event.rs

```diff
@@ -0,0 +1,272 @@
+#![allow(missing_docs)]
+
+use bevy_app::App;
+use bevy_ecs::prelude::*;
+use bevy_input::keyboard::KeyboardInput;
+use bevy_input::touch::TouchInput;
+use bevy_input::{
+    mouse::{MouseButtonInput, MouseMotion, MouseWheel},
+    touchpad::{TouchpadMagnify, TouchpadRotate},
+};
+use bevy_reflect::Reflect;
+use bevy_window::{
+    ApplicationLifetime, CursorEntered, CursorLeft, CursorMoved, FileDragAndDrop, Ime,
+    ReceivedCharacter, RequestRedraw, WindowBackendScaleFactorChanged, WindowCloseRequested,
+    WindowCreated, WindowDestroyed, WindowFocused, WindowMoved, WindowOccluded, WindowResized,
+    WindowScaleFactorChanged, WindowThemeChanged,
+};
+
+/// Wraps all `bevy_window` events in a common enum.
+///
+/// Read these events with `EventReader<WinitEvent>` if you need to
+/// access window events in the order they were received from `winit`.
```

Comment:
I would like to see a note about how each of these events *also* has its own dedicated event stream.